tests/notebooks/training-integration.ipynb

```diff
@@ -0,0 +1,904 @@
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client.delete_pytorchjob(name=PYTORCHJOB_NAME)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@retry(\n",
+    "    wait=wait_exponential(multiplier=2, min=1, max=10),\n",
+    "    stop=stop_after_attempt(30),\n",
+    "    reraise=True,\n",
+    ")\n",
+    "def assert_pytorchjob_removed(client, job_name):\n",
+    "    \"\"\"Wait for PyTorchJob be removed.\"\"\"\n",
+    "    # fetch the existing PyTorchJob names\n",
+    "    # verify that the Job was deleted successfully\n",
+    "    jobs = {job.metadata.name for job in client.list_pytorchjobs()}\n",
+    "    assert job_name not in jobs, f\"Failed to delete PyTorchJob {job_name}!\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# wait for PyTorch job to be removed successfully\n",
+    "assert_pytorchjob_removed(client, PYTORCHJOB_NAME)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test PaddlePaddle"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define a PaddleJob\n",
+    "\n",
+    "Define a PaddleJob object before deploying it. This PaddleJob is using [this](https://github.com/kubeflow/training-operator/blob/11b7a115e6538caeab405344af98f0d5b42a4c96/examples/paddlepaddle/simple-cpu.yaml) example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PADDLEJOB_NAME = \"paddle-simple-cpu\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "port = V1ContainerPort(container_port=37777, name=\"master\")\n",
+    "\n",
+    "container = V1Container(\n",
+    "    name=\"paddle\",\n",
+    "    image=\"registry.baidubce.com/paddlepaddle/paddle:2.4.0rc0-cpu\",\n",
+    "    command=[\"python\"],\n",
+    "    args=[\"-m\", \"paddle.distributed.launch\", \"run_check\"],\n",
+    "    ports=[port],\n",
+    ")\n",
+    "\n",
+    "replica_spec = V1ReplicaSpec(\n",
+    "    replicas=2,\n",
+    "    restart_policy=\"OnFailure\",\n",
+    "    template=V1PodTemplateSpec(\n",
+    "        metadata=V1ObjectMeta(annotations={\"sidecar.istio.io/inject\": \"false\"}),\n",
+    "        spec=V1PodSpec(containers=[container]),\n",
+    "    ),\n",
+    ")\n",
+    "\n",
+    "paddlejob = KubeflowOrgV1PaddleJob(\n",
+    "    api_version=\"kubeflow.org/v1\",\n",
+    "    kind=\"PaddleJob\",\n",
+    "    metadata=V1ObjectMeta(name=PADDLEJOB_NAME),\n",
+    "    spec=KubeflowOrgV1PaddleJobSpec(\n",
+    "        run_policy=V1RunPolicy(clean_pod_policy=\"None\"),\n",
```

Comment:
How did this come to be? It doesn't appear in the upstream example.

Reply:
the run_policy attribute must be set (raises error if None) when creating any kind of job using the API. Setting it this way with `clean_pod_policy=None` doesn't affect the run of the job, just used this one for consistency with the other job types. I'll change the comment to say the test is loosely based on the upstream example.